# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Hardik144/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [14]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

I choose the freestyle lane: understanding the relationship between search ranking position and click-through rate (CTR). I want to explore whether pages appearing at different average positions receive different levels of engagement. This is useful because a content team could use the findings to identify pages that may need further review. I chose this question because the starter dataset contains observable search performance metrics, and I have already explored the correlation between average position and CTR.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [15]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

Research question: Is average search position associated with the click-through rate of a page?

Decision: Which pages should a content team prioritize for further investigation and possible improvement?

Action: The content team could review pages with relatively low CTR, examine their content and search appearance, and decide whether further changes or experiments are appropriate.

Cost of a wrong recommendation: The team could waste time and resources changing pages that do not need improvement, or overlook pages that could benefit from attention. Unnecessary changes could also reduce page performance.

Unit of analysis: An individual page in the starter dataset.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [16]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


import pandas as pd
import numpy as np

# Load the FlyRank starter dataset
df = pd.read_csv("content_refresh_anonymized.csv")

# 1. Dataset size
print("Number of pages:", len(df))

# 2. Average click-through rate
print("Average CTR:", round(df["ctr"].mean(), 4))

# 3. Average position
print("Average position:", round(df["avg_position"].mean(), 2))

# 4. Correlation between position and CTR
correlation = df["avg_position"].corr(df["ctr"])
print("Correlation between average position and CTR:", round(correlation, 3))

Number of pages: 30000
Average CTR: 0.5107
Average position: 16.34
Correlation between average position and CTR: -0.073


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [17]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

I can describe observed associations between average search position and CTR in the available dataset. The dataset contains 3,000 pages, with an average CTR of 0.5101 and an average position of 16.34. The correlation between average position and CTR is -0.073, indicating a very weak negative linear association.

These results may help the content team investigate pages that could benefit from further review. However, correlation does not prove causation. I cannot claim that changing a page's position will necessarily increase CTR, predict Google's ranking decisions, or guarantee improvements. Further validation using a proper train/test split and additional evidence would be required.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.